### PR TITLE
Feature/87 centralise state handler

### PR DESCRIPTION
**Pull request by dami-gg**
#87

### CODE REVIEW

File: src/app/navigation.service.spec.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+    it('should dispatch an action to change the current product id', () => {
+      navigationService.loadCurrentContext(mock.applicationId, mock.productId);
+
+      expect(stateHandlerService.dispatch.calls.argsFor(1)).toEqual(
+        [mock.actions.changeCurrentProductId, mock.productId])
+    });
+
+    xit('should dispatch an action to show the applications menu if there is an application id', () => {
+      let result = navigationService.loadCurrentContext(mock.applicationId, mock.productId);
```
Comment (dami-gg) [inline src/app/navigation.service.spec.js]: I've struggled with this long enough today and still puzzles me but I didn't want it to be blocking me to finish this task. I'll try again soon in another task related to unit testing

Comment (gregdeane) [inline src/app/navigation.service.spec.js]: It's because `MOCKED_ID` at the bottom is set to `0`. Try setting to `1`.

**Comment by gregdeane** [comment]
👍

**Comment by cjhowald** [comment]
👍

**Pull request merged**